File: app/src/main/java/com/boardgamegeek/ui/TabActivity.java
package com.boardgamegeek.ui;

import android.os.Bundle;
import android.text.TextUtils;

import com.boardgamegeek.R;
import com.google.android.material.tabs.TabLayout;

import androidx.appcompat.app.ActionBar;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.viewpager.widget.ViewPager;
import butterknife.BindView;

/**
 * A navigation drawer activity that displays a view pager.
 */
public abstract class TabActivity extends DrawerActivity {
	@BindView(R.id.coordinator) CoordinatorLayout coordinator;
	@BindView(R.id.tab_layout) TabLayout tabLayout;
	@BindView(R.id.viewpager) ViewPager viewPager;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		final ActionBar actionBar = getSupportActionBar();
		if (actionBar != null) {
			actionBar.setDisplayHomeAsUpEnabled(true);
		}

		setUpViewPager();
		tabLayout.setupWithViewPager(viewPager);
	}

	protected abstract void setUpViewPager();

	@Override
	protected int getLayoutResId() {
		return R.layout.activity_tab;
	}

	protected void safelySetTitle(String title) {
		if (!TextUtils.isEmpty(title)) {
			ActionBar actionBar = getSupportActionBar();
			if (actionBar != null) {
				actionBar.setTitle(title);
			}
		}
	}
}

File: app/src/main/java/com/boardgamegeek/ui/TabActivity.kt
package com.boardgamegeek.ui

import android.os.Bundle
import android.view.View
import androidx.fragment.app.FragmentPagerAdapter
import com.boardgamegeek.R
import com.boardgamegeek.databinding.ActivityTabBinding

/**
 * A navigation drawer activity that displays a view pager.
 */
abstract class TabActivity : DrawerActivity() {
    protected lateinit var binding: ActivityTabBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        binding = ActivityTabBinding.inflate(layoutInflater)

        super.onCreate(savedInstanceState)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        binding.viewPager.adapter = createAdapter()
        binding.tabLayout.setupWithViewPager(binding.viewPager)
    }

    protected abstract fun createAdapter(): FragmentPagerAdapter

    override fun inflateRootView(): View {
        return binding.root
    }

    protected fun safelySetTitle(title: String?) {
        if (!title.isNullOrBlank()) {
            supportActionBar?.title = title
        }
    }
}
